lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -103,47 +105,21 @@ def identify(self, requirement_or_candidate):
         """
         return requirement_or_candidate.canonical_package_id
 
-    def get_preference(self, *args, **kwargs):
-        # type: (t.Any, t.Any) -> t.Union[float, int]
+    def get_preference(
+        self,
+        identifier: str,
+        resolutions: t.Mapping[str, Candidate],
+        candidates: t.Mapping[str, t.Iterator[Candidate]],
+        information: t.Iterator[t.NamedTuple],
```

Comment:
Looking at resolvelib, this could be more precise (imprecise types contribute to incomplete coverage). But that's something to tackle separately.